CommonLibSF/CMakeLists.txt

```diff
@@ -0,0 +1,166 @@
+cmake_minimum_required(VERSION 3.21)
+
+# singleton target across multiple projects
+if(TARGET CommonLibSF)
+	return() 
+endif()
+
+# options if not defined
+option(SFSE_SUPPORT_XBYAK "Enables trampoline support for Xbyak." ON)
+
+# standards & flags
+set(CMAKE_CXX_STANDARD 23)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
+set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_DEBUG OFF)
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake" CACHE FILEPATH "")
+
+# info
+project(
+	CommonLibSF
+	LANGUAGES CXX
+	VERSION 1.0.0
+)
+
+include(GNUInstallDirs)
+
+# out-of-source builds only
+if(${PROJECT_SOURCE_DIR} STREQUAL ${PROJECT_BINARY_DIR})
+	message(FATAL_ERROR "In-source builds are not allowed.")
+endif()
+
+# dependencies
+find_package(spdlog CONFIG REQUIRED)
+
+# source files
+execute_process(COMMAND powershell -ExecutionPolicy Bypass -File "${CMAKE_CURRENT_SOURCE_DIR}/cmake/make-sourcelist.ps1" "${CMAKE_CURRENT_SOURCE_DIR}" "${CMAKE_CURRENT_BINARY_DIR}")
+include(${CMAKE_CURRENT_BINARY_DIR}/sourcelist.cmake)
+source_group(
+	TREE ${CMAKE_CURRENT_SOURCE_DIR}
+	FILES ${SOURCES}
+)
+
+add_library(
+	${PROJECT_NAME}
+	STATIC
+	${SOURCES}
+	.clang-format
+)
+
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
+
+target_compile_definitions(
+	${PROJECT_NAME}
+	PUBLIC
+		WINVER=0x0601	# windows 7, minimum supported version by skyrim special edition
```

Comment:
Possibly push these to 0x0A00 for Windows 10 (min supported version of Starfield)

Reply:
Thanks, I didn't know that.
fixed: 217d6444b17edddad276c69f09374783dd56d980